### ISSUE TITLE
Add SIG names to Youtube uploads and organize them into playlists again

### ISSUE THREAD

**Issue by rauno56** [open-telemetry/community#715]
It seems that the videos used to have SIG name in them and they used to be filed under a SIG-specific playlist, but it stopped around October 2020. Could we bring this back? It's difficult to find relevant videos the way it is right now.

Example here: https://www.youtube.com/watch?v=QnbY2qxHBW0&list=PLVYDBkQ1Tdyw5ccPWGur1s2Ohnt9f6eEO

**Comment by mtwo** [comment]
Hey there! We used to upload the SIG recordings manually, which was very tedious but gave use the opportunity to name each one. Unfortunately this also consumed a few hours of people's time each week and also became infeasible with slow home internet connections and bandwidth limitations.

Unfortunately, Zoom doesn't know which meeting it's connected to or recording, meaning that our automation solution doesn't have this data to tag the YouTube upload with. I understand that the Kubernetes community (which uses the same CNCF Zoom solution) has the same issue (?).

**Comment by Rauno56** [comment]
Don't know how they do it, but they have more descriptive names.
Ex: [SIG Node meetings](https://www.youtube.com/playlist?list=PL69nYSiGNLP1wJPj5DYWXjiArF-MJ5fNG), [All meetings](https://www.youtube.com/c/KubernetesCommunity/videos)

The URL or ID of the call should reveal the meeting. However, I have no clue what Zoom API looks like, but I'm sure it is possible somehow! :)

**Comment by Rauno56** [comment]
Here are [Youtube Guidelines](https://github.com/kubernetes/community/blob/4026287dc3a2d16762353b62ca2fe4b80682960a/communication/youtube/youtube-guidelines.md) for k8s.

**Comment by mtwo** [comment]
Interesting, we're also using Splain on their recommendation. I wonder if they have one Zoom account for each SIG and thus can just name the uploads after the Zoom account that created them. This may be difficult with so many SIGs on OpenTelemetry, which is why we share three Zoom accounts across all of them.

@SergeyKanzhelev do you know if Kubernetes has dedicated Zoom accounts for each SIG? Is this how they're able to correctly name their meeting uploads?

**Comment by SergeyKanzhelev** [comment]
Yes, in k8s each sig gets a zoom account and responsible for managing it. Videos are uploaded automatically and leads have to manually start recording in the beginning (I saw cases and personally was forgetting it), review, rename, and publish recordings. Also host is required to start every meeting - they wouldn't start without the host presence. Which makes at least one lead participation in every meeting mandatory. Ad-hoc record-able meetings are not trivial to organize, you need a lead to participate. In k8s every SIG also maintains the google calendar for meeting times. A new SIG or working group creation is a heavy process.

We have as many SIGs as k8s (21 in OT vs. 24 in k8s). But the scale - number of people and leads is not proportional. We have significantly less. So managing this will not be easy. I am sure we can get there over time.

As for immediate steps we can think of - we can either extend automation to try to guess the meeting based on recording time and the room name. We can create an automation that will rename recordings based on comment in YouTube. So maintainers will need to make a comment. Or do a thing similar to what we did with the private Governance meeting. Since it's private, but recordable, it is explicitly created in one of the zoom accounts. And automation knows how to categorize it: https://github.com/open-telemetry/community/blob/b24dd0e4ab9b4391190dd20a12bf68aff2007162/tools/OpenTelemetryYoutube/OpenTelemetryYoutube/Program.cs#L145-L146 (there are issues with this approach and it is a bit fragile. We can scale it if somebody is ready to play a role of zoom maintainer with the weekly time commitment).

If there is interest to address this problem, I can advice on what can be done and what drawbacks would be.

**Comment by Rauno56** [comment]
> As for immediate steps we can think of - we can either extend automation to try to guess the meeting based on recording time and the room name.

Using the room name/ID would make sense imho although, I must be missing something if k8s requires separate accounts to make it happen.

**Comment by XSAM** [comment]
Any process on this one? I have to iterate every SIG record that happened on the same day to confirm which one I want to watch. 😢

**Comment by matschaffer** [comment]
I found https://github.com/open-telemetry/community/blob/main/docs/how-meeting-recordings-upload-works.md from @alanwest on the CNCF slack today which definitely helps provide some insight into why this might not be working at this stage. 

I'd also love to see it come back. As someone mostly non-timezone aligned with the meetings I rely on the recordings to keep up to date and just flip through all the recordings till I find the right one. If the organizer screenshares the agenda this is easier, but sometimes they don't and I have to just listen and guess at which one I'm listening to.

**Comment by Rauno56** [comment]
At this point I'd be even happy to find, rename and categorize our(JS) SIG meetings by hand every week.

To really make this useful for people we should have separate channels for different SIGs because no-one wants to subscribe to a channel with so much irrelevant content, but that's another topic.

There's also an [old channel](https://www.youtube.com/channel/UCVF98rxLfV3sv8Oi4fcJ1GQ) - do we know who manages that and whether we can move the content over somehow?

**Comment by Mario-Hofstaetter** [comment]
> To really make this useful for people we should have separate channels for different SIGs because no-one wants to subscribe to a channel with so much irrelevant content

This

**Comment by alanwest** [comment]
I'm the one that has the tool running regularly that makes the videos public. As mentioned previously in this thread, the videos are originally named after the name of the Zoom account used for the meeting - for example, here's a snippet of recent videos uploaded to YouTube

<img width="324" alt="Screen Shot 2022-05-19 at 4 49 31 PM" src="https://user-images.githubusercontent.com/3676547/169422162-547982e7-d31d-46c1-b48c-30e92bda01df.png">

So, renaming the videos to include the date it was uploaded from Zoom is a tiny step up in usability but still far from ideal.

I too want a better experience finding the videos I'm interested in. It'd be great if it could be automated. It's tedious to do this manually and far too easy to forget.

I'd be happy to improve on the tool, but I'd like to find the best way to go about it. The primary problem to solve is to automate the mapping of a video to a specific SIG. I'd imagine this might be done using the Google Calendar API, Zoom API or a combination of the two.

@SergeyKanzhelev is this something I could discuss with you when you're free? Seems you've thought a bit about this.

**Comment by tylerbenson** [comment]
I also noticed this issue again last week.  Do the videos have indication of which zoom link was used?  I wonder if we had unique zoom links for each SIG if that would make the process of putting into playlists easier...

**Comment by dyladan** [comment]
> I wonder if we had unique zoom links for each SIG if that would make the process of putting into playlists easier...

At one point @mtwo implemented this but it caused other problems (I don't remember the exact issues) and we reverted back to the shared link scenario we have today.

**Comment by mtwo** [comment]
Unfortunately there aren't many good options here :(

We currently have four or five Zoom accounts that are provided by the CNCF, and we use each account's instant meeting link to schedule our regular SIG meetings. This has two downsides: we can't host simultaneous meetings from the same Zoom account (hence why we have several accounts), and Zoom doesn't know the name of each meeting (because they're not actually scheduled with Zoom).

Last year we tried to go around this by actually scheduling meetings with a single Zoom account (using it as the organizer). This had the benefit of giving each recording the correct name, however Zoom started blocking our meetings as they overlapped and because the host never joined, so we had to revert back to our current system.

The only alternative that I see would be creating Zoom accounts for each SIG, however the CNCF likely won't pay for that, as we have a huge number of SIGs.
